lua/telescope/pickers.lua

```diff
@@ -429,12 +429,14 @@ function Picker:find()
     self:set_prompt(self.default_text)
   end
 
-  if vim.tbl_contains({ "insert", "normal" }, self.initial_mode) then
+  if vim.tbl_contains({ "insert", "normal", "select" }, self.initial_mode) then
     local mode = vim.fn.mode()
     local keys
     if self.initial_mode == "normal" then
       -- n: A<ESC> makes sure cursor is at always at end of prompt w/o default_text
       keys = mode ~= "n" and "<ESC>A<ESC>" or "A<ESC>"
+    elseif self.initial_mode == "select" then
+      keys = mode ~= "n" and [[<ESC>^v$<C-g>]] or [[<ESC>^v$<C-g>]]
```

Comment:
Is it github or is the lhs and rhs of this "if" the same?

Reply:
Ah thanks, threw this together in 10mins with a beer when I realized I had two PRs to reply to after 1.5 months :sweat_smile: mode switching worked as intended, but I'll check switching from normal to select mode once more to make sure.